File: app/src/main/java/com/u1tramarinet/livedatasample/NonNullSupplier.java
package com.u1tramarinet.livedatasample;

import androidx.annotation.NonNull;

import java.util.function.Supplier;

/**
 * Supplier that guarantee NonNull value.
 * @param <T> type.
 */
@FunctionalInterface
interface NonNullSupplier<T> extends Supplier<T> {
    @NonNull
    @Override
    T get();
}

File: app/src/main/java/com/u1tramarinet/livedatasample/NonNullSupplier.kt
package com.u1tramarinet.livedatasample

import java.util.function.Supplier

/**
 * Supplier that guarantee NonNull value.
 * @param <T> type.
</T> */
@FunctionalInterface
internal interface NonNullSupplier<T> : Supplier<T> {
    override fun get(): T
}